### PR TITLE
OAuth2Manager sample for WinAppSDK 1.7

### PR DESCRIPTION
**Pull request by akanpatel2206**
## Description

This pull request adds a sample application demonstrating the use of the OAuth2Manager API in WinAppSDK. The sample showcases how to perform OAuth2.0 for authorization code an dimplicit grant type. It includes implementations in C++ and C# to highlight cross-language compatibility. The sample aims to provide developers with a practical example of integrating OAuth2.0 functionality into desktop apps.

## Target Release

1.7

## Checklist

- [x] Samples build and run using the Visual Studio versions listed in the [Windows development docs](https://docs.microsoft.com/windows/apps/windows-app-sdk/set-up-your-development-environment?tabs=stable#2-install-visual-studio).
- [x] Samples build and run on all supported platforms (x64, x86, ARM64) and configurations (Debug, Release).
- [x] Samples set the minimum supported OS version to Windows 10 version 1809.
- [x] Samples build clean with no warnings or errors.
- [x] **[For new samples]**: Samples have completed the [sample guidelines checklist](https://github.com/microsoft/WindowsAppSDK-Samples/blob/main/docs/samples-guidelines.md#checklist) and follow [standardization/naming guidelines](https://github.com/microsoft/WindowsAppSDK-Samples/blob/main/docs/samples-guidelines.md#standardization-and-naming).

### CODE REVIEW

File: Samples/OAuth2Manager/cpp-winui/TestOAuthInCPlusPlus/MainWindow.xaml.cpp [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            Uri{ L"ms-testoauthincpp-launch://" });
+
+        outputTextBlock().Text(L"Implicit OAuth with redirect performed : Auth res: " + res.ResponseUri().RawUri());
+    }
+
+    winrt::Windows::Foundation::IAsyncAction MainWindow::authCodeGrantWithRedirectButton_Click(IInspectable const&, RoutedEventArgs const&)
+    {
+         
+		auto clientID = L"Ov23liNoEsMlGXgQPLcC";
+        auto clientSecret = L"1237a9e0534937990d08f1b0304773e11c21c05b";
```
Comment (godlytalias) [inline Samples/OAuth2Manager/cpp-winui/TestOAuthInCPlusPlus/MainWindow.xaml.cpp:46]: May be better to put dummy values here?!

File: Samples/OAuth2Manager/cpp-winui/TestOAuthInCPlusPlus/StartProgram.cpp [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+using namespace Microsoft::UI::Dispatching;
+using namespace Microsoft::Windows::AppLifecycle;
+using namespace Windows::ApplicationModel::Activation;
+using namespace winrt::TestOAuthInCPlusPlus::implementation;
+
+static Microsoft::UI::Dispatching::DispatcherQueue dispatcherQueue{ nullptr };
+
+void OnActivated(Windows::Foundation::IInspectable const&, AppActivationArguments const& args)
+{
+    if (args.Kind() == ExtendedActivationKind::Protocol)
```
Comment (godlytalias) [inline Samples/OAuth2Manager/cpp-winui/TestOAuthInCPlusPlus/StartProgram.cpp:24]: May be to good to add a check here for the Uri to confirm that Protocol activation is happening from OAuth, may need a specific Uri for OAuth as well

File: Samples/OAuth2Manager/cpp-winui/TestOAuthInCPlusPlus/MainWindow.xaml.cpp (hunk 1 above)
Comment (akanpatel2206) [reply Samples/OAuth2Manager/cpp-winui/TestOAuthInCPlusPlus/MainWindow.xaml.cpp]: yes! Thanks :) 
Done.

File: Samples/OAuth2Manager/cpp-winui/TestOAuthInCPlusPlus/Package.appxmanifest [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+<Package
+  xmlns="http://schemas.microsoft.com/appx/manifest/foundation/windows10"
+  xmlns:mp="http://schemas.microsoft.com/appx/2014/phone/manifest"
+  xmlns:uap="http://schemas.microsoft.com/appx/manifest/uap/windows10"
+  xmlns:rescap="http://schemas.microsoft.com/appx/manifest/foundation/windows10/restrictedcapabilities"
+  IgnorableNamespaces="uap rescap">
+
+  <Identity
+    Name="c1ee0edc-ac81-4a80-8dc3-1c882490ab2b"
+    Publisher="CN=ayushjaiswal"
```
[the remark this answers is not in the document]
Comment (akanpatel2206) [reply Samples/OAuth2Manager/cpp-winui/TestOAuthInCPlusPlus/Package.appxmanifest]: Done.

File: Samples/OAuth2Manager/cpp-winui/TestOAuthInCPlusPlus/MainWindow.xaml.cpp [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+            dispatcherQueue.TryEnqueue([this, accessToken, authRequestResult]()
+                {
+                    outputTextBlock().Text(L"Auth code type TokenResponse : Token: " + accessToken + L"\nAuth res: " + authRequestResult.ResponseUri().RawUri());
+                });
+        }
+    }
+
+    void MainWindow::OnUriCallback(Uri const& responseUri)
+    {
+        if (!OAuth2Manager::CompleteAuthRequest(responseUri))
```
Comment (hiteshwpfmsft) [inline Samples/OAuth2Manager/cpp-winui/TestOAuthInCPlusPlus/MainWindow.xaml.cpp:71]: Should we log an error here?

File: Samples/OAuth2Manager/cpp-winui/TestOAuthInCPlusPlus/StartProgram.cpp (hunk 2 above)
Comment (akanpatel) [reply Samples/OAuth2Manager/cpp-winui/TestOAuthInCPlusPlus/StartProgram.cpp:24]: Done.

**Review by godlytalias — APPROVED** [review]

File: Samples/OAuth2Manager/cpp-winui/TestOAuthInCPlusPlus/MainWindow.xaml.cpp (hunk 4 above)
Comment (akanpatel2206) [reply Samples/OAuth2Manager/cpp-winui/TestOAuthInCPlusPlus/MainWindow.xaml.cpp:71]: Left a comment for the client to decide the message.

**Pull request merged**